plugins/unbound/unbound_

```diff
@@ -0,0 +1,535 @@
+#!/bin/sh
+# -*- sh -*-
+
+: << =cut
+
+=head1 NAME
+
+unbound_ - Munin plugin to monitor the Unbound DNS resolver.
+
+=head1 APPLICABLE SYSTEMS
+
+System with unbound daemon.
+
+=head1 CONFIGURATION
+
+Environment variables for this script
+
+	unbound_conf	- where the unbound.conf file is located.
+	unbound_control	- where to find unbound-control executable.
+	spoof_warn	- what level to warn about spoofing
+	spoof_crit	- what level to crit about spoofing
+
+You can set them in your munin/plugin-conf.d/plugins.conf file
+with:
+
+  [unbound_*]
+  user root
+  env.unbound_conf /etc/unbound/unbound.conf
+  env.unbound_control /usr/sbin/unbound-control
+  env.spoof_warn 1000
+  env.spoof_crit 100000
+
+Use the .env settings to override the defaults.
+
+To install; enable statistics and unbound-control in unbound.conf
+	server:		extended-statistics: yes
+			statistics-cumulative: no
+			statistics-interval: 0
+	remote-control:	control-enable: yes
+Run the command unbound-control-setup to generate the key files.
+
+=head1 USAGE
+
+This plugin can create different graphs depending on what name you link
+it as (with ln -s) into the plugins directory. You can link it multiple
+times. If you are only a casual user, the _hits and _by_type are most
+interesting, possibly followed by _by_rcode.
+
+unbound_hits		- base volume, cache hits, unwanted traffic
+unbound_queue		- to monitor the internal requestlist
+unbound_memory		- memory usage
+unbound_by_type		- incoming queries by type
+unbound_by_class	- incoming queries by class
+unbound_by_opcode	- incoming queries by opcode
+unbound_by_rcode	- answers by rcode, validation status
+unbound_by_flags	- incoming queries by flags
+unbound_histogram	- histogram of query resolving times
+
+=head1 AUTHOR
+
+Copyright 2008 W.C.A. Wijngaards
+          2013 Kim B. Heino
+
+=head1 LICENSE
+
+BSD
+
+=head1 MAGIC MARKERS
+
+ #%# family=auto
+ #%# capabilities=autoconf
+
+=cut
+
+. ${MUNIN_LIBDIR}/plugins/plugin.sh
+
+state="${MUNIN_PLUGSTATE}/munin-plugin-unbound.state"
+seentags="${MUNIN_PLUGSTATE}/munin-plugin-unbound_seentags.state"
+conf=${unbound_conf:-/etc/unbound/unbound.conf}
+ctrl=${unbound_control:-/usr/sbin/unbound-control}
+warn=${spoof_warn:-1000}
+crit=${spoof_crit:-100000}
+lock=$state.lock
+
+# number of seconds between polling attempts.
+# makes the statefile hang around for at least this many seconds,
+# so that multiple links of this script can share the results.
+lee=55
+
+# to keep things within 19 characters
+ABBREV="-e s/total/t/ -e s/thread/t/ -e s/num/n/ -e s/query/q/ -e s/answer/a/ -e s/unwanted/u/ -e s/requestlist/ql/ -e s/type/t/ -e s/class/c/ -e s/opcode/o/ -e s/rcode/r/ -e s/edns/e/ -e s/mem/m/ -e s/cache/c/ -e s/mod/m/"
+
+# get value from $1 into return variable $value
+get_value ( ) {
+	value="`grep '^'$1'=' $state | sed -e 's/^.*=//'`"
+	if test "$value"x = ""x; then
```

Comment:
Maybe `[ -z "$value" ]` would be more readable?
(I think, the habit of prepending or appending a fixed string (*x*) derives from the traditional lack of quoting applied by shell programmers ...)